mithril-stm/src/stm.rs

```diff
@@ -705,35 +779,35 @@ impl<D: Clone + Digest + FixedOutput + Send + Sync> StmAggrSig<D> {
         msg: &[u8],
         avk: &StmAggrVerificationKey<D>,
         parameters: &StmParameters,
-    ) -> Result<(), StmAggregateSignatureError<D>> {
-        let mut nr_indices = 0;
-        let mut unique_indices = HashSet::new();
-        let mut leaves = Vec::new();
-
-        for (sig, reg_party) in &self.signatures {
-            let msgp = avk.mt_commitment.concat_with_msg(msg);
-            sig.check_indices(parameters, &reg_party.1, &msgp, avk)?;
-
-            for &index in &sig.indexes {
-                unique_indices.insert(index);
-                nr_indices += 1;
-            }
+    ) -> Result<(Vec<Signature>, Vec<VerificationKey>), StmAggregateSignatureError<D>> {
```

Comment:
I see why this new function signature is useful. However, this change should be reflected in the documentation of the function.